### PR TITLE
[Test Proxy] Add error response when start test proxy

### LINKED ISSUE #26138: [Test Proxy] Add error response with debug information when startup hangs
While adding test proxy support to a new test suite, a user was having issues running tests -- `pytest` would correctly collect tests and the test proxy container would start, but things would hang at that point. Below is a snippet of test proxy container logs, which continued with many more "Recorded: 0 Played Back: 0" statements.
```
Dumping Resolved Configuration Values:
VERSION = 6.0.6
URLS = http://0.0.0.0:5000;https://0.0.0.0:5001
SYSTEM_GLOBALIZATION_INVARIANT = true
RUNNING_IN_CONTAINER = true
PATH = /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
NO_AT_BRIDGE = 1
Logging =
Logging:LogLevel =
Logging:LogLevel:Microsoft.Hosting.Lifetime = Information
Logging:LogLevel:Microsoft = Information
Logging:LogLevel:Default = Information
Logging:Console =
Logging:Console:FormatterName =
HOSTNAME = ffcfcc8294da
HOME = /root
ENVIRONMENT = Development
DOTNET_VERSION = 6.0.6
DOTNET_SYSTEM_GLOBALIZATION_INVARIANT = true
DOTNET_RUNNING_IN_CONTAINER = true
contentRoot = /proxyserver
CERT_IMPORT_SH = apply-dev-cert.sh
CERT_FOLDER = /var/certwork
ASPNET_VERSION = 6.0.6
ASPNETCORE_URLS = http://0.0.0.0:5000;https://0.0.0.0:5001
ASPNETCORE_ENVIRONMENT = Development
applicationName = Azure.Sdk.Tools.TestProxy
AllowedHosts = *
[19:11:30] warn: Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository[60]
      Storing keys in a directory '/root/.aspnet/DataProtection-Keys' that may not be persisted outside of the container. Protected data will be unavailable when container is destroyed.
[19:11:30] info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[62]
      User profile is available. Using '/root/.aspnet/DataProtection-Keys' as key repository; keys will not be encrypted at rest.
[19:11:30] info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[58]
      Creating key {614a8dbe-fbb3-4fcd-924a-1b8d4e273571} with creation date 2022-09-09 19:11:30Z, activation date 2022-09-09 19:11:30Z, and expiration date 2022-12-08 19:11:30Z.
[19:11:30] warn: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[35]
      No XML encryptor configured. Key {614a8dbe-fbb3-4fcd-924a-1b8d4e273571} may be persisted to storage in unencrypted form.
[19:11:30] info: Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository[39]
      Writing data to file '/root/.aspnet/DataProtection-Keys/key-614a8dbe-fbb3-4fcd-924a-1b8d4e273571.xml'.
[19:11:30] info: Microsoft.Hosting.Lifetime[14]
      Now listening on: http://0.0.0.0:5000/
[19:11:30] info: Microsoft.Hosting.Lifetime[14]
      Now listening on: https://0.0.0.0:5001/
[19:11:30] info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
[19:11:30] info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Development
[19:11:30] info: Microsoft.Hosting.Lifetime[0]
      Content root path: /proxyserver
[19:11:31] Recorded: 0  Played Back: 0
[19:11:32] Recorded: 0  Played Back: 0
```
Normally, before the "Recorded: 0 Played Back: 0" logs, there should be logs of the recording/playback session starting request.

The issue turned out to be that the test proxy devcert was never set up correctly, so our attempts to [check proxy availability](https://github.com/Azure/azure-sdk-for-python/blob/408ee52e5e183832bb5421c73fedaa6fed0f81fc/tools/azure-sdk-tools/devtools_testutils/proxy_startup.py#L143) failed. Ideally, a certificate issue should be detected and reported straight away, instead of having to wait for attempts to time out. At the very least, we should report the failure upon timeout with a helpful message that suggests double-checking certificate setup, any local setting for the `PROXY_URL` environment variable, etc. that could have caused the issue.

### PR DESCRIPTION
**Pull request by ponopono0322**
# Description

Fix #26138.

If provide the `-o log_cli=true` option at the starting the test proxy,
example:
```
pytest tests/ -o log_cli=true
```

you can get the log:
```
ERROR    root:proxy_startup.py:95 Please check 'SSL_CERT_DIR' path
ERROR    root:proxy_startup.py:97 Please check 'REQUESTS_CA_BUNDLE' path
```

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: tools/azure-sdk-tools/devtools_testutils/proxy_startup.py [hunk 1]
```diff
@@ -87,6 +87,17 @@ def check_availability() -> None:
...
 
+def check_system_proxy_availability() -> None:
+    """Checks for SSL_CERT_DIR and REQUESTS_CA_BUNDLE environment variables."""
+    ssl_cert = "SSL_CERT_DIR"
+    ca_bundle = "REQUESTS_CA_BUNDLE"
+
+    if not os.environ.get(ssl_cert):
+        _LOGGER.error(f"Please check '{ssl_cert}' path")
+    if not os.environ.get(ca_bundle):
+        _LOGGER.error(f"Please check '{ca_bundle}' path")
```
Comment (mccoyp) [inline tools/azure-sdk-tools/devtools_testutils/proxy_startup.py:98]: Nitpicking to make the error message a bit more exact:
```suggestion
    if not os.environ.get(ssl_cert):
        _LOGGER.error(f"Please ensure the '{ssl_cert}' environment variable is correctly set in your test environment")
    if not os.environ.get(ca_bundle):
        _LOGGER.error(f"Please ensure the '{ca_bundle}' environment variable is correctly set in your test environment")
```

**Review by mccoyp — COMMENTED** [review]
Thank you so much for opening this PR! I'm so sorry I didn't see this earlier, since I didn't get automatically added as a reviewer. I just have a small wording change suggestion, but this is a great change -- thanks again!

File: tools/azure-sdk-tools/devtools_testutils/proxy_startup.py (hunk 1 above)
Comment (ponopono0322) [reply tools/azure-sdk-tools/devtools_testutils/proxy_startup.py]: Never mind, it's ok! Your suggestion is much better.

**Comment by ponopono0322** [comment]
By the way, this doesn't seem to be a perfect solution. Can I continue like this?

File: tools/azure-sdk-tools/devtools_testutils/proxy_startup.py [hunk 2]
```diff
@@ -87,6 +87,17 @@ def check_availability() -> None:
...
 
 
+def check_system_proxy_availability() -> None:
+    """Checks for SSL_CERT_DIR and REQUESTS_CA_BUNDLE environment variables."""
+    ssl_cert = "SSL_CERT_DIR"
+    ca_bundle = "REQUESTS_CA_BUNDLE"
+
+    if not os.environ.get(ssl_cert):
+        _LOGGER.error(f"Please ensure the '{ssl_cert}' environment variable is correctly set in your test environment")
+    if not os.environ.get(ca_bundle):
```
Comment (mccoyp) [inline tools/azure-sdk-tools/devtools_testutils/proxy_startup.py:97]: After thinking about this again, I realized that we only need these environment variables set when we're using an SSL connection. That's true by default, but if someone sets [PROXY_URL](https://github.com/Azure/azure-sdk-for-python/blob/5fd52b9ee039f8711322bd7ea43af763d326291a/tools/azure-sdk-tools/devtools_testutils/config.py#L8) to be something like `"http://localhost:5000"`, we shouldn't raise error logs about certificate setup
```suggestion
    if PROXY_URL.startswith("https") and not os.environ.get(ssl_cert):
        _LOGGER.error(f"Please ensure the '{ssl_cert}' environment variable is correctly set in your test environment")
    if PROXY_URL.startswith("https") and not os.environ.get(ca_bundle):
```

**Review by mccoyp — APPROVED** [review]
Sorry about the late response again! 🙁 I think this is actually a good solution -- I considered if we should raise an actual exception instead of logging errors, but I think it's better to not block users in case they have a unique configuration. Minimizing erroneous logs by ensuring our proxy endpoint actually needs a certificate will improve this, but it's still a great change in my opinion!

File: tools/azure-sdk-tools/devtools_testutils/proxy_startup.py (hunk 2 above)
Comment (ponopono0322) [reply tools/azure-sdk-tools/devtools_testutils/proxy_startup.py]: Thank you for your thorough review!

**Comment by ponopono0322** [comment]
Thank you for your approval @mccoyp . I tried to merge it myself, but it comes out as below. May I ask for the approval of the merge?
Screenshot 2022-11-09 at 10 25 25 AM

**Comment by mccoyp** [comment]
Hi @ponopono0322, I was out of office last week but am just trying this on my end before merging to make sure it's all in order 🙂

**Pull request merged**